Prometheus/TextSerializer.cs

```diff
@@ -15,16 +15,26 @@ internal sealed class TextSerializer : IMetricsSerializer
         private static readonly byte[] LeftBrace = { (byte)'{' };
         private static readonly byte[] RightBraceSpace = { (byte)'}', (byte)' ' };
         private static readonly byte[] Space = { (byte)' ' };
+        private static readonly byte[] SpaceHashSpaceLeftBrace = { (byte)' ', (byte)'#', (byte)' ', (byte)'{' };
         private static readonly byte[] PositiveInfinity = PrometheusConstants.ExportEncoding.GetBytes("+Inf");
-
-        public TextSerializer(Stream stream)
+        private static readonly byte[] NegativeInfinity = PrometheusConstants.ExportEncoding.GetBytes("-Inf");
+        private static readonly byte[] NotANumber = PrometheusConstants.ExportEncoding.GetBytes("NaN");
+        private static readonly byte[] PositiveOne = PrometheusConstants.ExportEncoding.GetBytes("1.0");
+        private static readonly byte[] Zero = PrometheusConstants.ExportEncoding.GetBytes("0.0");
+        private static readonly byte[] NegativeOne = PrometheusConstants.ExportEncoding.GetBytes("-1.0");
+        private static readonly byte[] EofNewLine = PrometheusConstants.ExportEncoding.GetBytes("# EOF\n");
+
+        public TextSerializer(Stream stream, ExpositionFormat fmt = ExpositionFormat.Text)
```

Comment:
@sandersaares happy to refactor this now that this is fleshed out. I do need some guidance on the primitives to use.